tensorboard/webapp/metrics/effects/index.ts

```diff
@@ -224,17 +224,17 @@ export class MetricsEffects implements OnInitEffects {
 
   private readonly visibleCardsWithoutDataChanged$ = this.actions$.pipe(
     ofType(actions.cardVisibilityChanged),
-    switchMap(() => this.getVisibleCardFetchInfos().pipe(take(1))),
-    map((fetchInfos) => {
+    withLatestFrom(this.getVisibleCardFetchInfos()),
```

Comment:
Also switching to use more idiomatic rxjs - getVisibleCardFetchInfos is derived from ngrx state so it will always have a value ready. Using withLatestFrom instead of take(1) shouldn't change behavior.

This is a side effect of my investigation into the problem. I could move it into a different PR if you'd like.